osu.Game/Rulesets/Scoring/HealthProcessor.cs

```diff
@@ -86,13 +102,12 @@ private bool meetsAnyFailCondition(JudgementResult result)
             if (CheckDefaultFailCondition(result))
                 return true;
 
-            if (FailConditions != null)
+            foreach (var condition in Mods.Value.OfType<IHasFailCondition>().OrderByDescending(m => m.RestartOnFail))
```

Comment:
1. Do we really need the `OrderByDescending`?
2. Let's try to not use Linq in potentially hot spots where performance could matter, like on every judgement. I don't know if we've enforced this up until now, but I think it's good to start being wary of it.

Reply:
I got an idea for that